src/plugins/github/porcelain.test.js

```diff
@@ -27,184 +30,212 @@ describe("GitHub porcelain", () => {
-      const reviews = pr.reviews();
-      expect(reviews).toHaveLength(2);
-      const comments = reviews[0].comments();
-      expect(comments).toHaveLength(1);
-      const comment = comments[0];
-      expect(comment.parent()).toEqual(reviews[0]);
-      expect(comment.url()).toBe(
-        "https://github.com/sourcecred/example-github/pull/5#discussion_r171460198"
+    it("have reviews", () => {
+      expectPropertiesToMatchSnapshot(prs, (e) =>
+        e.reviews().map((r) => r.url())
       );
-      expect(comment.body()).toBe("seems a bit capricious");
-      expect(comment.authors().map((a) => a.login())).toEqual(["wchargin"]);
-      expect(comment).toEqual(asEntity(graph, comment.address()));
     });
+  });
 
-    it("Comments", () => {
-      const issue = issueOrPRByNumber(6);
-      const comments = issue.comments();
-      expect(comments.length).toMatchSnapshot();
-      const comment = comments[0];
-      expect(comment.type()).toBe(COMMENT_NODE_TYPE);
-      expect(comment.body()).toBe("A wild COMMENT appeared!");
-      expect(comment.url()).toBe(
-        "https://github.com/sourcecred/example-github/issues/6#issuecomment-373768442"
+  describe("pull request reviews", () => {
+    const reviews = pullRequest.reviews();
+    it("have review comments", () => {
+      expectPropertiesToMatchSnapshot(reviews, (e) =>
+        e.comments().map((e) => e.url())
       );
-      expect(comment.node()).toMatchSnapshot();
-      expect(comment.address()).toEqual(comment.node().address);
-      expect(comment.authors().map((x) => x.login())).toEqual(["decentralion"]);
-      expect(comment).toEqual(asEntity(graph, comment.address()));
-      expect(comment.parent()).toEqual(issue);
     });
+    it("have states", () => {
+      expectPropertiesToMatchSnapshot(reviews, (e) => e.state());
+    });
+  });
 
-    it("Authors", () => {
-      const authors = porcelain.authors();
-      // So we don't need to manually update the test if a new person posts
-      expect(authors.length).toMatchSnapshot();
-
-      const decentralion = authors.find((x) => x.login() === "decentralion");
-      if (decentralion == null) {
-        throw new Error("Who let the lions out?");
-      }
-      expect(decentralion.url()).toBe("https://github.com/decentralion");
-      expect(decentralion.type()).toBe(AUTHOR_NODE_TYPE);
-      expect(decentralion.subtype()).toBe("USER");
-      expect(decentralion.node()).toMatchSnapshot();
-      expect(decentralion.address()).toEqual(decentralion.node().address);
-      expect(decentralion).toEqual(asEntity(graph, decentralion.address()));
+  describe("asEntity", () => {
+    it("works for each wrapper", () => {
+      allWrappers.forEach((w) => {
+        expect(asEntity(w.graph, w.address())).toEqual(w);
+      });
+    });
+    it("errors when given an address with the wrong plugin name", () => {
+      const addr: Address = {
+        pluginName: "the magnificent foo plugin",
```

Comment:
Good to see that we haven't lost the magnificent foo plugin in the great reorganization.